src/Collection.php

```diff
@@ -205,10 +205,6 @@ public function aggregate(array $pipeline, array $options = [])
             $options['readConcern'] = $this->readConcern;
         }
 
-        if ( ! isset($options['typeMap']) && ( ! isset($options['useCursor']) || $options['useCursor'])) {
-            $options['typeMap'] = $this->typeMap;
-        }
-
```

Comment:
We only need to get ride of the `useCursor` logic here, as we sill want to inherit the type map if it's not set in `$options`.